components/controls/ClipboardButton.tsx

```diff
@@ -65,25 +66,41 @@ export default class ClipboardButton extends React.PureComponent<Props, State> {
     }
   }
 
-  showTooltip = () => {
+  handleMouseEnter = () => {
     if (this.mounted) {
       this.setState({ tooltipShown: true });
+    }
+  };
+
+  handleMouseLeave = () => {
+    if (this.mounted) {
+      this.setState({ tooltipShown: false });
+    }
+  };
+
+  showTooltip = () => {
+    if (this.mounted) {
+      this.setState({ isCopied: true, tooltipShown: true });
       setTimeout(() => {
         if (this.mounted) {
-          this.setState({ tooltipShown: false });
+          this.setState({ isCopied: false, tooltipShown: false });
         }
       }, 1000);
     }
   };
 
   render() {
     return (
-      <Tooltip overlay={translate('copied_action')} visible={this.state.tooltipShown}>
+      <Tooltip
+        onMouseEnter={this.handleMouseEnter}
+        onMouseLeave={this.handleMouseLeave}
```

Comment:
I'm really skeptical about exposing those internal stuff... I think we should never have to care about that when using the tooltip.

I think it would be way simpler to play with the visible state, switching it between `true` to force it visible when copied and `undefined` to have the default mouseover behavior.

Don't hesitate to ping me if you want to discuss further the implem.

Reply:
Let's not forget what the original intent was: to show a confirmation message when clicked (after click, it displays a tooltip with "Copied to Clipboard" for 1s). Now, we're adding a tooltip to explain what it does as well...

Probably that line 97 is sufficient:

```ts
overlay={translate(this.state.isCopied ? 'copied_action' : 'copy_to_clipboard')}
```

That way, we leave the enter/leave logic in Tooltip (as it should, completely agree with Greg), and the message is still displayed correctly once clicked.